strapi/strapi
# fix(data-transfer): Adjust MYSQL Codes to catch foreign key errors

### What does it do?

Adjust MYSQL Codes to catch foreign key errors

| `e.code` is non numeric |
|:-:|
| ![image](https://github.com/strapi/strapi/assets/1272897/cfe4ce43-5d6b-455b-a508-6c9f9eca6f51) |


### Why is it needed?

The data transfer does not work with MYSQL DB

### How to test it?

Do a transfer with orphans in your MYSQL database (create an entity with a relation to another content type and in the database delete one of the ends of the relation, leaving the row linking both content types in the DB). Then execute a transfer to another Strapi project and observe that there is a warning in the server about the foreign key constraint, but the transfer continues as expected.

### Related issue(s)/PR(s)

https://github.com/strapi/strapi/pull/19186
